/*
 * Copyright 2025 ByteChef
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.bytechef.platform.configuration.workflow;

/**
 * Consulted before an editor surface writes a workflow definition by a caller-supplied id, allowing modules to refuse
 * edits to workflows they own and regenerate themselves.
 *
 * @author Ivica Cardic
 */
public interface WorkflowUpdateGuard {

    /**
     * Throws when the workflow must not be edited directly.
     *
     * @param workflowId the id of the workflow about to be updated
     */
    void checkUpdatable(String workflowId);
}
